mpldatacursor/pick_info.py

```diff
@@ -111,20 +112,15 @@ def line_props(event):
         return dict(x=xorig[i], y=yorig[i])
 
     # ax.step is actually implemented as a Line2D with a different drawstyle...
-    drawstyle = event.artist.get_drawstyle()
     xs_data = xorig[max(i - 1, 0) : i + 2]
     ys_data = yorig[max(i - 1, 0) : i + 2]
-    if drawstyle == "steps_pre":
-        xs_data = _interleave(xs_data, xs_data[:-1])
-        ys_data = _interleave(ys_data, ys_data[1:])
-    elif drawstyle == "steps_post":
-        xs_data = _interleave(xs_data, xs_data[1:])
-        ys_data = _interleave(ys_data, ys_data[:-1])
-    elif drawstyle == "steps_mid":
-        mid_xs = (xs_data[:-1] + xs_data[1:]) / 2
-        xs_data = _interleave(xs_data, np.column_stack([mid_xs, mid_xs]))
-        ys_data = _interleave(
-            ys_data, np.column_stack([ys_data[:-1], ys_data[1:]]))
+    drawstyle_conv = {
+        "_draw_lines": lambda xs, ys: (xs, ys),
+        "_draw_steps_pre": cbook.pts_to_prestep,
```

Comment:
I agree, but I'd rather target something a touch older than 1.5.  I think
things actually worth with 1.0 right now, but that's excessive.  1.3 is
probably a better bet.

Let me double check what the various LTS distros currently have in their
repos.

Incidentally, my apologies for being incommunicado for the past bit. I've
been travelling more or less non-stop for the past week.

On Tue, May 31, 2016 at 6:58 PM, Antony Lee <notifications@github.com>
wrote:

> In mpldatacursor/pick_info.py
> <https://github.com/joferkington/mpldatacursor/pull/67#discussion_r65283364>
> :
>
> >      xs_data = xorig[max(i - 1, 0) : i + 2]
> >      ys_data = yorig[max(i - 1, 0) : i + 2]
> > -    if drawstyle == "steps_pre":
> > -        xs_data = _interleave(xs_data, xs_data[:-1])
> > -        ys_data = _interleave(ys_data, ys_data[1:])
> > -    elif drawstyle == "steps_post":
> > -        xs_data = _interleave(xs_data, xs_data[1:])
> > -        ys_data = _interleave(ys_data, ys_data[:-1])
> > -    elif drawstyle == "steps_mid":
> > -        mid_xs = (xs_data[:-1] + xs_data[1:]) / 2
> > -        xs_data = _interleave(xs_data, np.column_stack([mid_xs, mid_xs]))
> > -        ys_data = _interleave(
> > -            ys_data, np.column_stack([ys_data[:-1], ys_data[1:]]))
> > +    drawstyle_conv = {
> > +        "_draw_lines": lambda xs, ys: (xs, ys),
> > +        "_draw_steps_pre": cbook.pts_to_prestep,
>
> @joferkington <https://github.com/joferkington> I can obviously restore
> the previous implementation, but perhaps that's a good time to pick a
> minimal version of matplotlib to support? I see a few back-compatibility
> snippets here and there in the codebase.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/joferkington/mpldatacursor/pull/67/files/bb93efa860f54daed1665c479e8595949dc52b70#r65283364>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA3WMwi5ktFSmyxOClYxm_dUZZL2owNMks5qHMsvgaJpZM4IpVY7>
> .
>